crates/dojo-world/src/metadata.rs

```diff
@@ -373,6 +374,26 @@ impl ArtifactMetadata {
 
         Ok(response.hash)
     }
+    pub fn calculate_ipfs_hash(&self) -> std::io::Result<String> {
```

Comment:
To test this function, you should do the following:

1. Run a simple migration without changes to the contracts.
2. Check the IPFS CID that are generated and printed by sozo.
3. Take the content of such files and write a unit test to verify you're computing it correctly.